content/docs/guides/gke/gcp-e2e.md

```diff
@@ -0,0 +1,1009 @@
+
+    **Note:** The `--build-arg` flags in the `docker build` command
+    pass the arguments into the Dockerfile, which then passes them on to the
+    Python program.
+
+    If everything went well, your program is now encapsulated in a new
+    container.
+
+1. Test the container locally:
+
+    ```
+    docker run -it ${TRAIN_IMG_PATH}
+    ```
+
+    You may see some warnings from TensorFlow about deprecated functionality.
+    Then you should see training logs start appearing in your output, similar
+    to these:
+
+    ```
+    step 0/2000, training accuracy 0.14
+    step 100/2000, training accuracy 0.8
+    step 200/2000, training accuracy 0.94
+    step 300/2000, training accuracy 0.94
+    step 400/2000, training accuracy 0.96
+    step 500/2000, training accuracy 0.94
+    step 600/2000, training accuracy 0.94
+    step 700/2000, training accuracy 0.88
+    step 800/2000, training accuracy 0.98
+    step 900/2000, training accuracy 0.88
+    step 1000/2000, training accuracy 0.84
+    step 1100/2000, training accuracy 0.96
+    step 1200/2000, training accuracy 0.98
+    step 1300/2000, training accuracy 0.94
+    step 1400/2000, training accuracy 0.98
+    step 1500/2000, training accuracy 1
+    step 1600/2000, training accuracy 0.96
+    step 1700/2000, training accuracy 0.96
+    step 1800/2000, training accuracy 0.96
+    step 1900/2000, training accuracy 0.96
+    ```
+
+1. When you see log entries similar to those above, your model training is
+   working. You can terminate the container with **Ctrl+c**. Don't worry if
+   you see an error saying that your service account doesn't
+   have GET access on the bucket. That access is not necessary at this point.
+
+Next, upload the container image to Container Registry so that you can
+run it on your Kubernetes Engine cluster.
+
+1. Run the following command to authenticate to Container Registry:
+
+    ```
+    gcloud auth configure-docker
+    ```
+
+1. Push the container to Container Registry:
+
+    ```
+    docker push ${TRAIN_IMG_PATH}
+    ```
+    The push may take a few minutes to complete. You should see Docker progress 
+    updates in your command window.
+
+1. Wait until the process is complete, then you should see your new container
+   image listed on the [Container Registry page][gcp-container-registry]
+   on the GCP console.
+
+<a id="train-model"></a>
+### Train the model on Kubernetes Engine
+
+Now you are ready to run the TensorFlow training job on your cluster on
+Kubernetes Engine.
+
+1. Use the [`ks generate`][ks-generate] command to generate a ksonnet component
+   from the [`tf-job` prototype][tf-job-prototype]. The code below generates a
+   component called `train1`:
+
+    ```
+    cd ${HOME}/kubeflow-introduction/${DEPLOYMENT_NAME}_ks_app
+    ks generate tf-job-operator train1
+    ```
+
+    You should now see a new file, `train1.jsonnet`, in your
+    `${DEPLOYMENT_NAME}_ks_app/components/` directory.
```

Comment:
Hmmm, I took it out in response to this comment, but I've put it back again. The ksonnet config aspect of Kubeflow is unfamiliar to many people, and they may not know where to look for a 'components' directory. If this were part of a command, I'd agree that we could remove the extra bit of path, but since it's in some explanatory text my 2c is it's clearer to leave it in.